File: 2022-2023/sourcecode/lab/lab4/JetpackNavigationJava/app/src/main/java/pl/edu/uwr/pum/jetpacknavigationjava/FragmentB.java
package pl.edu.uwr.pum.jetpacknavigationjava;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class FragmentB extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_b, container, false);

        TextView textView = view.findViewById(R.id.textViewB);
        textView.setText(String.valueOf(
                getArguments() != null ? getArguments().getInt("key") : 0
        ));

        view.findViewById(R.id.fabB).setOnClickListener(v ->{
            NavDirections action = FragmentBDirections.toFragmentA();
            Navigation.findNavController(view).navigate(action);
        });

        return view;
    }
}

File: 2022-2023/sourcecode/lab/lab4/JetpackNavigationKotlin/app/src/main/java/pl/edu/uwr/pum/jetpacknavigationkotlin/FragmentB.kt
package pl.edu.uwr.pum.jetpacknavigationkotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.navigation.Navigation
import com.google.android.material.floatingactionbutton.FloatingActionButton

class FragmentB : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_b, container, false)

        view.findViewById<TextView>(R.id.textViewB).text = arguments?.getInt("key").toString()

        view.findViewById<FloatingActionButton>(R.id.fabB).setOnClickListener {
            val action = FragmentBDirections.toFragmentA()
            Navigation.findNavController(view).navigate(action)
        }

        return view
    }
}